snapshots/overlay/overlay.go

```diff
@@ -102,22 +117,57 @@ func NewSnapshotter(root string, opts ...Opt) (snapshots.Snapshotter, error) {
 		return nil, err
 	}
 
-	if err := os.Mkdir(filepath.Join(root, "snapshots"), 0700); err != nil && !os.IsExist(err) {
-		return nil, err
+	firstSnapshot := true
+	if err := os.Mkdir(filepath.Join(root, "snapshots"), 0700); err != nil {
+		if !os.IsExist(err) {
+			return nil, err
+		}
+		firstSnapshot = false
 	}
 	// figure out whether "userxattr" option is recognized by the kernel && needed
 	userxattr, err := overlayutils.NeedsUserXAttr(root)
 	if err != nil {
 		logrus.WithError(err).Warnf("cannot detect whether \"userxattr\" option needs to be used, assuming to be %v", userxattr)
 	}
 
+	var (
+		metacopyKernel bool
+		metacopyOn     *bool
+	)
+	if config.metacopyOn != nil {
+		if metacopyKernel, err = overlayutils.SupportsMetacopy(); err != nil && !os.IsExist(err) {
```

Comment:
why `!os.IsExist` here? (Should it be IsNotExist?) And shouldn't we treat it as "failed to check for kernel option" as well?

Reply:
>And shouldn't we treat it as "failed to check for kernel option" as well?

It's reasonable, if a user specified metacopy option then we expect that there's support from the kernel,  otherwise just error out. I'll fix this in the next iteration. Thanks!